src/main/java/com/gluonhq/substrate/util/Strings.java

```diff
@@ -108,4 +112,46 @@ public static String randomString(int targetStringLength) {
                 .toString();
     }
 
+    /**
+     * Removes duplicated strings from a sourceList list, keeping the items with higher index
+     * in the list. Any item in the keep list will not be removed.
```

Comment:
that is not guaranteed to be the correct approach. If an entry appear twice, there are a number of possible scenario's. Both may be needed, first may be needed, last may be needed.

Reply:
There is a keep list for strings that might be duplicated and needed. 
So far the case I found is the whole/no-whole combination. 
I've tested this PR in all possible platforms, and there was no issue removing the rest of the duplicates.